### DISCUSSION TITLE
Will thanos Thanos Compactor slow down the query latency

### DISCUSSION THREAD

**Discussion by yliu138repo** [thanos-io/thanos#7051] [Questions & Answers]
Hi team,

I am new to Thanos. We are wondering that if the Thanos Compactor will slow down the query process? 

I know that Thanos Compactor will reduce the number of blocks and compact index indexces. Is there any other process behind the scenes like compressing the data stored? If yes, when query occurs, would it need any extra steps/process to uncompress the data from object storage, which could introcude extra latency?

Thanks a lot.

**Comment by MichaHoffmann** [comment]
A well-running compactor is very important for the query process. It doesnt "compress" data as much as it merges the indexes and samples of multiple blocks into one, reducing the amount of blocks that storage gateway has to query. Since its a separate component that only works on object storage its actions have no effect on current queries at all.

**Reply by yliu138repo** [reply]
Great that is the one we are after. Thanks Michael
